--- work/ffi/build.py.orig	2018-08-08 05:09:45.626074900 -0500
+++ work/ffi/build.py	2018-08-08 05:38:57.426021600 -0500
@@ -53,7 +53,11 @@
     # seems a bit lacking here.
     cmake_dir = os.path.join(here_dir, 'dummy')
     # LLVM 4.0+ needs VS 2015 minimum.
-    for generator in ['Visual Studio 14 2015']:
+    generators = []
+    if os.environ.get("CMAKE_GENERATOR"):
+        generators.append(os.environ.get("CMAKE_GENERATOR"))
+    generators.append('Visual Studio 14 2015')
+    for generator in generators:
         if is_64bit:
             generator += ' Win64'
         build_dir = tempfile.mkdtemp()
