diff --git a/ffi/CMakeLists.txt b/ffi/CMakeLists.txt
index 87cf00c..b8a99f1 100755
--- a/ffi/CMakeLists.txt
+++ b/ffi/CMakeLists.txt
@@ -7,6 +7,7 @@ project(llvmlite_ffi)
 include(CheckIncludeFiles)
 
 find_package(LLVM REQUIRED CONFIG)
+find_package("ZLIB" REQUIRED)
 
 message(STATUS "Found LLVM ${LLVM_PACKAGE_VERSION}")
 message(STATUS "Using LLVMConfig.cmake in: ${LLVM_DIR}")
@@ -47,4 +48,6 @@ llvm_map_components_to_libnames(llvm_libs all)
 list(REMOVE_ITEM llvm_libs "OptRemarks")
 
 # Link against LLVM libraries
+include_directories(${ZLIB_INCLUDE_DIRS})
 target_link_libraries(llvmlite ${llvm_libs})
+target_link_libraries(llvmlite ${ZLIB_LIBRARIES})
