diff --git a/llvmlite/binding/ffi.py b/llvmlite/binding/ffi.py
index 3f53b2a..333a313 100644
--- a/llvmlite/binding/ffi.py
+++ b/llvmlite/binding/ffi.py
@@ -119,6 +119,11 @@ if os.name == 'nt':
     # Append DLL directory to PATH, to allow loading of bundled CRT libraries
     # (Windows uses PATH for DLL loading, see http://msdn.microsoft.com/en-us/library/7d83bc18.aspx).  # noqa E501
     os.environ['PATH'] += ';' + _lib_dir
+    import sys
+    if sys.version_info >= (3, 8):
+        _prefix = os.path.normpath(sys.prefix)
+        os.add_dll_directory(_prefix)
+        os.add_dll_directory(os.path.join(_prefix, 'Library', 'bin'))
 
 
 _lib_name = get_library_name()
@@ -146,6 +151,10 @@ for _lib_path in _lib_paths:
     try:
         lib = ctypes.CDLL(_lib_path)
     except OSError:
+        from traceback import print_exc
+        print_exc()
+        from subprocess import run
+        run(['dumpbin', '/dependents', _lib_path])
         continue
     else:
         break
